modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2379,14 +2379,20 @@ def explode(self, column):
     def apply(self, func, axis, *args, **kwargs):
```

Comment:
Potentially, instead of handling the logic in `_callable_func` do we want to create another method like `_callable_func_{on|with}_series` and call that from apply instead?

Reply:
That might be cleaner/more readable.